facp.md

```diff
@@ -0,0 +1,224 @@
+
+and finally we need to check if it enabled the ACPI. It can be done by 3 steps:
+
+1. For the PM1a CNT Block: Could it work?
+```c
+int i;
+for(i = 0; i < 300; i++){
+	if(inw((uint16_t) PM1a_CNT_BLK & (uint16_t) SCI_EN) == 1){
+		break;
+	}
+		// If else, you can sleep a bit and continue again!!
+}
+```
+2. Do the same thing for the pm2b cnt block.
+3. Give error if not, give success message if is.
+
+### II. ACPI initialization
+
+You must have found the FACP and have the header.
+
+S5 Address can be found by several lines of code basically doing a while
+
+But you need to first add 36 from the DSDT as the address, get the dsdt length as the following code, and finally do a while until DSDT Length -- is greater than 0
+The code below performs a check at every byte, looking for a sequence of 4 characters, you should compare the first 4 characters via memcmp like functions and when you find it, you'll break and continue to the next step. Don't forget to add 1 every time to the address or you will not get the loop properly.
+```c
+	uint8_t* s5_addr = (uint8_t*) facp->dsdt + 36;
+	int dsdt_len = *(facp->dsdt + 1) - 36;
+	while(0 < dsdt_len--){
+		if(memcmp(s5_addr, "_S5_", 4)==0){
+			break;
+		}
+		s5_addr++;
+	}
+```
+Now that you have found the S5 addr then you will go make the init acpi function.
+
+All the stuff explained below will happen given that the DSDT length is greater than 0, which mean it has any data so you might want to get a while for this thing.
+
+First you need to check if AML is valid.
+
+You need to check if everything is in the right place.
+
+Your check to make it valid needs information below:
+
+| Encoding name | Encoding value | 
+| --------------|----------------|
+|   NameOp    |  0x08        |
+|   Byteprefix    |  0x0A        |
+|   Byteprefix    |  0x12        |
+
+Figure 1
+
+|               | PackageOP	 | Package Length | NumElements	   | Byteprefix#    | Byteprefix#    | Byteprefix#    | Byteprefix#    |
+| --------------|----------------|----------------|----------------|----------------|----------------|----------------|----------------|
+|   Data        |                |                |                | SLP_TYPa       | SLP_TYPb       | Res.           | Res.           |
+|   Enc. Val.   |                |                |  0x04          |                |                |                |                |
+
+Figure 2
+
+Figure 1 and 2 explains the encoding value of the encoding name and how all these look like repectively as a photo.
+
+An example of the explaination:
+```c
+if((*(s5_addr - 1) == 0x08 || ( *(s5_addr - 2) == 0x08 && *(s5_addr - 1) == '\\') ) && *(s5_addr + 4) == 0x12){
+	s5_addr += 5;
+	s5_addr += ((*s5_addr & 0xC0) >> 6) + 2;
+	
+	if (*s5_addr == 0x0A){
+		s5_addr++;
+	}
+	SLP_TYPa = *(s5_addr) << 10;
+	s5_addr++;
+			
+	if (*s5_addr == 0x0A){
+		s5_addr++;
+	}
+	SLP_TYPb = *(s5_addr) << 10;
+```
+
+Next, make all the defined data the same as the one of the FACP.
```

Comment:
i think there is something odd with markdown syntax. BEcause it is not marked as normal text. 

Btw leave always one empty line between text and code blocks. (btw i'm not sure that is the problem in this case..)

Reply:
The missing extra line is the problem it seems..